sockeye/inference.py

```diff
@@ -97,16 +97,20 @@ def _build_modules(self):
         # Encoder symbol & module
         source = mx.sym.Variable(C.SOURCE_NAME)
         source_length = mx.sym.Variable(C.SOURCE_LENGTH_NAME)
+        source_encoded_length = mx.sym.Variable(C.SOURCE_ENCODED_LENGTH_NAME)
```

Comment:
I am not sure I understand why you need to create another input to the graph here. You are overwriting source_encoded_length immediately below in the inner function (using nonlocal). As far as I can tell this variable symbol will not be part of the graph anyway so why not just set it to None?